pxtsim/runtime.ts

```diff
@@ -989,19 +989,16 @@ namespace pxsim {
             if (Runtime.messagePosted) Runtime.messagePosted(data);
         }
 
-        static postScreenshotAsync(opts?: SimulatorScreenshotMessage): Promise<void> {
+        static async postScreenshotAsync(opts?: SimulatorScreenshotMessage): Promise<void> {
```

Comment:
I think we should supply a default thumbnail, or have a stronger call to action to create one. The current default behavior doesn't feel great:
![image](https://user-images.githubusercontent.com/12176807/188215456-bf62a109-46b1-4163-bb59-417992e325ae.png)

If the user hasn't provided a screenshot, can we take one automatically three seconds into gameplay?

Reply:
the plan is to do exactly that, i just didn't do it as part of this PR. we used to do this but i think it broke at some point.

my basic plan is this:
1. try to take a screenshot after x seconds provided the screen isn't blank
2. when the user starts interacting with the sim, record frames for a gif (but put off encoding until they attempt to share)